test/system/config/test.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -eu -o pipefail
+
+# test compile and execution of a cupla program using its header-only variant
+# $1 C++ compiler name
+
+test_code_dir=$(dirname $0)
+
+# compiles the test case for the given accelerator
+# $1 flag if the compiled binary should be executed, 1 == execute binary else do not test binary 
+# $2 compiler name
+# $3 accelerator name (CpuOmp2Blocks, CpuOmp2Threads, CpuSerial, CpuThreads, GpuCudaRt, GpuHipRt)
+# $4 additional compiler flags
+function compile {
+    execute_bin=$1
+    compiler_name=$2
+    acc_name=$3
+    if [ $# -eq 4 ] ; then
+        compiler_flags="$4"
+    fi
+    ret=$(${compiler_name} ${test_code_dir}/main.cpp ${test_code_dir}/kernel.cpp  -I${test_code_dir}/../../../include -I${test_code_dir}/../../../alpaka/include -std=c++11 -DCUPLA_ACC_${acc_name} -DCUPLA_HEADER_ONLY -o ${acc_name} ${compiler_flags} && \
+        { echo 0; } || { echo 1; })
+    if [ $ret -eq 0 ] && [ $execute_bin -eq 1 ] ; then
+        ret=$(./$acc_name && { echo 0; } || { echo 1; })
+    fi
+    if [ $ret -ne 0 ] ; then
+        echo "Config header test failed for accelerator '$acc_name' with compiler flags '$compiler_flags'" >&2
+    fi
+    return $ret
+}
+
+boost_include="-I$BOOST_ROOT/include"
+
+compile 1 "$1" CpuOmp2Blocks "-fopenmp $boost_include"
```

Comment:
- while the compiler is a user-provided argument, the flags are compiler-specific
- linking fails for gcc here, because of a missing "-lstdc++" flag

Reply:
shouldn't the compiler be `g++` instead of `gcc` ?

Reply:
Travis reports `gcc` because I named the toolchain `gcc` for the tesst `CXX` is used.

@tdd11235813 on which system do you get issues when you not add `-lstdc++`. The CI tests are passing expect the travis crashs^^

Reply:
hrr of course, thx. Also tested intel compiler icc (icc accepts '-fopenmp'). xlc might behave differently, cannot test at the moment due to maintenance.

Reply:
Yes , when we move the CI to our gitlab and can test the more compiler and will solve the issues. Normally this is a tasks for cmake but we like to test without CMake.

Reply:
For this PR I will keep the hard codes test script. I will refactor this part and use CMake without pulling cupla/alpaka cmake in. So we will get support for all compiler and can still test the configurations.